src/batch/src/task/context.rs

```diff
@@ -67,6 +68,52 @@ pub trait BatchTaskContext: Clone + Send + Sync + 'static {
     fn mem_usage(&self) -> usize;
 
     fn create_executor_mem_context(&self, executor_id: &str) -> Option<MemoryContextRef>;
+    fn get_stop_flag(&self) -> Arc<StopFlag>;
+
+    fn get_stop_flag_ref(&self) -> &StopFlag;
+}
+
+/// We already have `shut_down` sender to stop execution.
+/// `StopFlag` used in executor to let it stop execute more timely when it can't receive message
+/// timely.
```

Comment:
So, after this PR, does the `shut_down` channel still make sense? Shall it be removed?

Reply:
Yes. StopFlag is just to prevent executor miss the stop signal when it don't await in processing datachunk. 
I will add this complement in comment.

Reply:
BTW, check StopFlag require atomic operation so it may have a impact on performance. So I think that it's only necessary to introduce it in the executor potentially use more memory (like join) to prevent problem described in #9418. For other executor, maybe they can just use shut_down channel?

cc @liurenjie1024

Reply:
I don't think it's a big issue. CPU is never the bottleneck.

BTW, I think `relaxed` would be enough here, if you worry about performance.

```rust
pub fn check_stop(&self) -> bool {
    // NOTE: Used `store(Ordering::Release),load(Ordering::Acquire)` to guarantee that the stop
    // reason can be viewed if the stop flag is visible.
    self.stop_flag.load(Ordering::Acquire)
}
```

Reply:
> So, after this PR, does the shut_down channel still make sense? Shall it be removed?

This is just first part of the process of stopping all executors using flag, so we can't remove it.

Reply:
> BTW, I think relaxed would be enough here, if you worry about performance.

`relaxed` will cause the concurrency problem because it can't guarantee cross thread happen-before relation🤔.

Reply:
> relaxed will cause the concurrency problem because it can't guarantee cross thread happen-before relation

It's acceptable that some `stop` may be executed a few loops later.